### ISSUE TITLE
Add pufferfish support.

### ISSUE THREAD

**Issue by Greninja9257** [macarooni-man/auto-mcs#108]
Title. (type of server)

**Comment by macarooni-man** [comment]
Hi @Greninja9257, thanks for your suggestion!

Adding a new distribution takes a lot of work and I need to see if it's possible to implement it.

Can you help me by trying to find an API where I can request a Minecraft version, and have it return a .jar file, or a URL from which I can retrieve one?

**Comment by Greninja9257** [comment]
Use MC Utils. You can download all the Pufferfish versions. Or you can manually download the desired server jars from their official downloads page.

**Comment by Greninja9257** [comment]
https://github.com/flytegg/mc-utils

**Comment by Greninja9257** [comment]
Also, official downloads I found it: https://pufferfish.host/downloads

**Comment by Greninja9257** [comment]
https://github.com/pufferfish-gg/Pufferfish?tab=readme-ov-file
I believe this might be related (IDK actually)
Building
./gradlew build
Or building a Paperclip JAR for distribution:

./gradlew paperclip

**Comment by macarooni-man** [comment]
Hi, @Greninja9257

Thanks for providing the download links and when I looked I found these, but I specifically requested the API URL for the place that actually serves files, not just the download link on their user-facing site. I need to be able to pull from their file service programmatically, and that's not something I can do with the URL's you provided.

I'm really busy trying to finish some other stuff right now and I don't have the time to inspect web requests and try to find that URL myself, which is why I'm asking for some help here.

I'm happy to accept a PR if you'd like to add this yourself, or otherwise provide the information I need to complete this.

**Comment by Greninja9257** [comment]
I found it! https://ci.pufferfish.host/

**Comment by macarooni-man** [comment]
Thanks for providing the detailed resources and info @Greninja9257! I'll see if it's possible to implement this in the next update and follow up

**Comment by macarooni-man** [comment]
Hi @Greninja9257, I've looked a lot into this and I'm hesitant about integrating Pufferfish in general. The issues for me are as follows:

- Closed source
- Not frequently updated (still on 1.21.3, Dec. 4th)
- I'm not sure I can trust the owner of the project considering how they've handled some situations in the past, especially with it being closed source
- The API is challenging to integrate

Additionally, Purpur seems to be more optimized in general and it's already implemented into auto-mcs. It doesn't have any of the other issues above either:

- [Open source](https://github.com/PurpurMC/Purpur)
- Very frequently updated (9 hours ago)
- Well, it's already implemented

I'm curious to hear your feedback on this, is there a specific reason Pufferfish is important to you? Maybe I'm out of the loop and there's something I'm missing that makes it much better than other solutions, but as it stand, I see the implementation as redundant and partially a risk.

**Comment by Greninja9257** [comment]
Pufferfish is fast for large servers and supports almost all plugins (add ons). The server fps was pretty high when I used it in the tradition server hosting technique.

**Comment by macarooni-man** [comment]
> Pufferfish is fast for large servers and supports almost all plugins (add ons). The server fps was pretty high when I used it in the tradition server hosting technique.

Yes, I understand that, but it also feels like you didn't read or consider anything that I said.

[Purpur integrates all of the same patches/features as Pufferfish](https://docs.pufferfish.host/other/pufferfish-vs-purpur/#what-is-purpur)

With this in mind, is there anything advantages you know of which Pufferfish provides over Purpur?

Also I'm not sure what you mean by "server fps" as that's usually a client-side metric. Did you mean TPS? Because it would be effectively identical to Purpur

**Comment by Greninja9257** [comment]
Here "choose Pufferfish if you want a regular, high performance server, Purpur if you want the extra features"
